<launch>
    <!--   DEFAULTS  -->
    <!--    export LIBGL_ALWAYS_SOFTWARE=1-->
    <arg name="is_sim" default="false" />
    <arg name="plot_show" default="false" />
    <arg name="plot_save" default="true" />
    <arg name="debug_with_images" default="true" />
    <arg name="publish_markers" default="true" />
    <arg name="debug" default="3" />
    <arg name="rasa" default="true" />
    <!--    <arg name="publish_markers" default="true" />-->

    <param name="plot_show" value="$(arg plot_show)" />
    <param name="plot_save" value="$(arg plot_save)" />
    <param name="debug" value="$(arg debug)" />
    <param name="rasa" value="$(arg rasa)" />
    <param name="publish_markers" value="$(arg publish_markers)" />
    <param name="debug_with_images" value="$(arg debug_with_images)" />
    <!--    <param name="publish_markers" value="$(arg publish_markers)" />-->

     <!--     TASK  -->
<!--     <node name="detect_objects_v8" pkg="lasr_object_detection_yolo" type="detect_objects_v8.py" output="screen" required="true"/>-->



    <!--     STATIC POINTS  -->

    <group if="$(arg is_sim)">
        <rosparam command="load" file="$(find lift)/config/final_lift_sim.yaml" />
    </group>
    <group unless="$(arg is_sim)">
        <rosparam command="load" file="$(find lift)/config/final_lift.yaml" />
    </group>


    <!--      PERCEPTION     -->
    <!--    <include file="$(find lasr_perception_server)/launch/perception_server.launch"/>-->
    <!--    <node name="lasr_object_detection_yolo" pkg="lasr_object_detection_yolo" type="yolo_object_detection_server.py" output="screen" respawn="true"/>-->
    <node pkg="lasr_object_detection_yolov8" type="service" name="service" output="screen"/>

    <include file="$(find lasr_object_detection_yolov8)/launch/service.launch">
        <arg name="debug" value="true" />
        <arg name="preload" value="['yolov8n-seg.pt']" />
    </include>


    <node pkg="lasr_perception_server" name="perception_server" type="perception_server.py"/>
    <node name="tf_module" pkg="tf_module" type="tf_transforms_base.py" output="screen" respawn="true"/>


</launch>
